src/main/java/store/config/InitialDataLoader.java

```diff
@@ -0,0 +1,106 @@
+package store.config;
+
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayDeque;
+import java.util.List;
+import java.util.Queue;
+import store.error.ErrorMessage;
+import store.error.exception.InvalidFailInitialDataLoadException;
+import store.product.domain.Product;
+import store.product.domain.ProductInfo;
+import store.product.domain.ProductRepository;
+import store.promotion.domain.Promotion;
+import store.promotion.domain.PromotionDateTime;
+import store.promotion.domain.PromotionRepository;
+import store.promotion.domain.PromotionType;
+import store.util.StringParser;
+
+public class InitialDataLoader {
+
+    private static final String PATH_PROMOTIONS = "promotions.md";
+    private static final String PATH_PRODUCTS = "products.md";
+    private static final String DELIMITER_COMMA = ",";
+    private final PromotionRepository promotionRepository;
+    private final ProductRepository productRepository;
+    private Long promotionSequence = 1L;
+    private Long productSequence = 1L;
+
+    public InitialDataLoader(final PromotionRepository promotionRepository, final ProductRepository productRepository) {
+        this.promotionRepository = promotionRepository;
+        this.productRepository = productRepository;
+    }
+
+    public void initialize() {
+        saveAllPromotions();
+        saveAllProducts();
+    }
+
+    private void saveAllPromotions() {
+        final Queue<String> fileLines = readFileLines(PATH_PROMOTIONS);
+        if (fileLines.isEmpty()) {
+            return;
+        }
+        fileLines.poll();
+        while (!fileLines.isEmpty()) {
+            final String line = fileLines.poll();
+            savePromotion(line);
+        }
+    }
+
+
+    private void saveAllProducts() {
+        final Queue<String> fileLines = readFileLines(PATH_PRODUCTS);
+        if (fileLines.isEmpty()) {
+            return;
+        }
+        fileLines.poll();
+        while (!fileLines.isEmpty()) {
+            final String line = fileLines.poll();
+            saveProduct(line);
+        }
+    }
```

Comment:
p5

코드가 중복되기에 이렇게 해보시는 것도 하나의 방법이 될 것 같습니다. 
하지만 만약 미래에 각각의 파일 양식이나 초기화하는 방법이 상이하게 변경될 경우도 고려했다면 수달님 방식이 더 좋을 것 같습니다.

```suggestion
public void initialize() {
    saveAllData(PATH_PROMOTIONS, this::savePromotion);
    saveAllData(PATH_PRODUCTS, this::saveProduct);
}

private void saveAllData(String path, Consumer<String> saveFunction) {
    final Queue<String> fileLines = readFileLines(path);
    if (fileLines.isEmpty()) {
        return;
    }
    fileLines.poll(); 
    while (!fileLines.isEmpty()) {
        final String line = fileLines.poll();
        saveFunction.accept(line);
    }
}
```

Reply:
이거 외에도 중복코드가 많긴 해요 ㅎㅎ 

그리고 말씀하신것 처럼 중복된 경우는 일단 하나의 메서드로 구현하고, 파일 양식이 변할경우 그 때 나누는게 좋을것 같습니다!

좋은 피드백 감사해요 ㅎㅎ